roles/install_nextcloud/tasks/main.yml

```diff
@@ -122,3 +125,26 @@
   register: nc_indices_cmd
   changed_when: '"Done" not in nc_indices_cmd.stdout'
   when: nextcloud_install_db
+
+- name: "[Main] - Patch from the app 'user_saml' the file SAMLController.php."
+  when:
+    - nextcloud_patch_user_saml_app
+    - nc_available_apps.enabled['user_saml'] is defined or nc_available_apps.disabled['user_saml'] is defined
+  tags:
+    - patch_user_saml_app
+  vars:
+    pattern: '.*\$this->session->set\(.user_saml\.samlUserData.,\s+\$_SERVER\);.*'
+    file2patch: "{{ nextcloud_webroot + 'apps/user_saml/lib/Controller/SAMLController.php' }}"
+  block:
+    - name: "[Main] - Check if SAMLController.php can be patched."
```

Comment:
Hmm I'm missing context here.
Was the following tasks (blockinfile) not idempotent for this check needed to be made ?

Reply:
Thanks @aalaesar! This works around a behaviour of blockinfile that is not desirable in this case. The patch is only carried out when the pattern-line can be found in SAMLController.php. Despite this being in "insertbefore" the blockinfile module irrespective of whether or not the pattern is matched inserts the block in the file. If the pattern is not found it simply adds the block at the end of the file. That is why you need to make the blockinfile execution conditional on whether the pattern can be found. This is done by reading in the file and check for pattern-match in the "when"-conditional and execute blockinfile only when the pattern is present. So hope this explains this quirky implementation. In my personal code I would add a comment to the source. Here I would add a comment explaining that: "# blockinfile adds the block at the end of the file if pattern is not found. This is undesired here.". I can add this here as well if desired.

Reply:
Ok I understand  thank you.
so it's a kind of failsafe in case the culprit code for this behaviour is removed.
LGTM